src/ex.rs

```diff
@@ -167,6 +175,27 @@ pub fn define_(
 }
 
 impl Experiment {
+    pub fn validate(&self) -> Result<()> {
+        if self.toolchains[0] == self.toolchains[1] {
+            bail!("reusing the same toolchain isn't supported");
+        }
+
+        if self.rustflags.is_some()
+            && !self.toolchains[0].enable_rustflags
+            && !self.toolchains[1].enable_rustflags
```

Comment:
It just occurred to me - maybe it's worth making this a tuple, since we don't support more than one toolchain at the moment?

Reply:
Well, some parts of Crater are using `[Toolchain; 2]`, and some `Vec<Toolchain>`. We should really unify them...

I personally prefer the fixed-length array to the tuple though, since it allows iteration.